Cython/Includes/cpython/datetime.pxd

```diff
@@ -161,6 +167,14 @@ cdef inline object datetime_new(int year, int month, int day, int hour, int minu
 cdef inline object timedelta_new(int days, int seconds, int useconds):
     return PyDateTimeAPI.Delta_FromDelta(days, seconds, useconds, 1, PyDateTimeAPI.DeltaType)
 
+# Create a datetime object using DateTime CAPI classmethod.
+cdef inline object datetime_from_timestamp(float timestamp, object tz):
+    return PyDateTimeApi.DateTime_FromTimestamp(timestamp, tz)
```

Comment:
That's not using the right signature. According to `datetime.h`, it's called like this:
```c
    PyDateTimeAPI->DateTime_FromTimestamp( \
        (PyObject*) (PyDateTimeAPI->DateTimeType), args, NULL)
```
where `args` is a tuple. So, I guess, the equivalent in Cython would be
```suggestion
# Create a datetime object using DateTime CAPI classmethod.
cdef inline object datetime_from_timestamp(timestamp, tz=None):
    return PyDateTimeApi.DateTime_FromTimestamp(
        PyDateTimeApi.DateTimeType, (timestamp, tz) if tz is not None else (timestamp,), NULL)
```
That suggests that the signature of that function pointer (`(object, object, object)`) is incorrect. It should be `(PyTypeObject*, object, PyObject*)`. Which … looks ugly, but that's how it's used.

Similarly, the function below also wants an `args` tuple as argument, not a float. (And, BTW, `float` means C float, not Python float.)

It doesn't really feel like these will be much faster than a plain Python call. Probably still visible in really time critical code, I guess.

Reply:
> (And, BTW, float means C float, not Python float.)

 Is that why you removed the `float` static typing from timestamp? Shouldn't there be `int` then, or is it just better without typing?

The documentation specifies:
> Return the local date and time corresponding to the POSIX timestamp, such as is returned by [time.time()](https://docs.python.org/3/library/time.html#time.time).

> time.time() → float

Reply:
> Is that why you removed the float static typing from timestamp? Shouldn't there be int then, or is it just better without typing?

Since we're only using it as an object, we shouldn't require users to pass in a C value. They may already have an object on their side that we can just pass through. Or they may not, in which case the conversion will happen automatically on the caller side. Taking an object and passing it on is as fast as it gets.